Add launch screen for phone app

LasseRosenow opened the pull request:
## Summary
This PR adds a launch screen with the home assistant logo to the `phone` app.
This is the android part of home-assistant/frontend/pull/10569.
In the spoken PR I recently modified the apps launch screen so that it is more consistent within the app.
Now this PR is to match the launch screen of the app on android, so that the whole startup experience is as clean as possible.

## Screenshots
<table>
<tr><th>Android Launch Screen</th><th>HTML Launch Screen</th></tr>
<tr>
<td>

![Screenshot_2021-11-18-02-20-02-523_io homeassistant companion android debug](https://user-images.githubusercontent.com/10547444/142335156-b97ee5ca-29c3-4acd-9b2a-62aa89879487.jpg)

</td>
<td>

![Screenshot_2021-11-18-02-23-19-994_com android chrome](https://user-images.githubusercontent.com/10547444/142335151-ee627f37-716f-4c0d-b23a-6b0c2d454565.jpg)

</td>
</tr>
<tr>
<td>

![Screenshot_2021-11-18-03-01-30-382_io homeassistant companion android debug](https://user-images.githubusercontent.com/10547444/142337660-a6f22196-6f1d-49b7-a72f-d486da4b2b2f.jpg)

</td>
<td>

![Screenshot_2021-11-18-02-54-23-205_com android chrome](https://user-images.githubusercontent.com/10547444/142337715-390154ed-f316-4c17-b451-6ac16d9d2fc8.jpg)

</td>
</tr>
</table>

## Any other notes
- I renamed app_icon.jpg to app_icon_round.jpg because the default home assistant icon is not round
- I renamed colorActivityBackground to colorBackground to be more in line with the material color names
- The launch screen uses its own background colors to match the background color of the html frontend. The android apps background colors are different for their own reasons

dshokouhi commented:
Have you tested this on a device running Android 12?  When I launch the debug app in dark mode this new launch screen background is first white then switches to black for the welcome screen so that doesn't seem right. Also on Android 12 this behavior is already done so this might interfere with the default splash screen that appears because I see the correct behavior today.

This is what the launch experience looks like on current `master` branch changes: https://photos.app.goo.gl/XwxUkCzmBBWfvNQF9

This is what the launch experience looks like with the debug APK proposed: https://photos.app.goo.gl/CR16vPoWqSdbAU29A

Are you trying to change the screen when webview is loading after a user is already logged in?

LasseRosenow commented:
Okay so I forgot about the new splash screen stuff in android 12. So I changed the code for android 12 to only change the background color. So that the background color of the splash screen matches the background color of the HTML app.

The white background in dark mode was a small mistake by my, as I added the new launch screen background color in the end and forgot to change the dark color value. This is fixed now aswell.

dshokouhi reviewed (approved):
Looks good to me, tested it out with current release and latest dev branch changes. Thanks for cleaning this up!
